modules/caddyhttp/headers/headers.go

```diff
@@ -192,6 +192,16 @@ type RespHeaderOps struct {
 
 // ApplyTo applies ops to hdr using repl.
 func (ops HeaderOps) ApplyTo(hdr http.Header, repl *caddy.Replacer) {
+	// delete all fields before manipulating HTTP headers
```

Comment:
```suggestion
	// before manipulating headers in other ways, check if there
	// is configuration to delete all headers, and do that first
	// because if a header is to be added, we don't want to delete
	// it also
```